third_party/nvfuser/csrc/type_promotion.cpp

```diff
@@ -100,6 +100,16 @@ c10::ScalarType computeTypes(
       c10::isIntegralType(common_dtype, /*includeBool=*/true)) {
     common_dtype = c10::get_default_dtype_as_scalartype();
   }
+
+  // Some ops like nextafter are not implemented for non-float types
+  if (config.require_full_precision_promoted) {
+    TORCH_CHECK(
+        common_dtype == c10::ScalarType::Float ||
+            common_dtype == c10::ScalarType::Double,
```

Comment:
I am not sure that this is the right place to perform this check.

Reply:
Hmm. I got this wrong too. PyTorch actually supports bfloat16, just not float16 for this op. It was added with a manual implementation taken from musl: https://github.com/pytorch/pytorch/pull/61829/files#diff-ece04c31934b3504382e10ed3e9a69f03ffabd81ad1a2a890aab19b1642f53c0R120